File: frontend/src/types/game.ts
export type DisplayMode = 'EMBEDDED' | 'POPUP';

export interface GameMetrics {
  views: number;
  likes: number;
  rating: number;
}

export interface GameDocument {
  id: string;
  title: string;
  description: string;
  original_url: string;
  url?: string;
  embed_code?: string;
  thumbnail_url: string;
  creator_id: string;
  creator_email?: string;
  creator_name?: string;
  display_mode: DisplayMode;
  metrics: GameMetrics;
  tags: string[];
  created_at: string;
  qr_image_url?: string;
  cover_image_url?: string;
  pdf_drive_url?: string;
  pdf_title?: string;
}

export interface ScrapedMetadata {
  title: string;
  description: string;
  thumbnail_url: string;
  display_mode: DisplayMode;
  tags: string[];
  original_url: string;
  embed_code?: string;
}

export interface SubmitGamePayload {
  url: string;
  embed_code?: string;
  custom_title?: string;
  custom_description?: string;
  custom_thumbnail_url?: string;
  custom_tags?: string[];
  creator_id?: string;
  creator_email?: string;
  creator_name?: string;
  qr_image_url?: string;
  cover_image_url?: string;
  pdf_drive_url?: string;
  pdf_title?: string;
}

File: backend/src/models/game.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub enum DisplayMode {
    Embedded,
    Popup,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GameMetrics {
    pub views: u64,
    pub likes: u64,
    pub rating: f32,
}

impl Default for GameMetrics {
    fn default() -> Self {
        Self {
            views: 0,
            likes: 0,
            rating: 5.0,
        }
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GameDocument {
    pub id: String,
    pub title: String,
    pub description: String,
    pub original_url: String,
    pub thumbnail_url: String,
    pub creator_id: String,
    pub display_mode: DisplayMode,
    pub metrics: GameMetrics,
    pub tags: Vec<String>,
    pub created_at: String,
}

#[derive(Debug, Deserialize)]
pub struct SubmitGameRequest {
    pub url: String,
    pub creator_id: Option<String>,
    pub custom_title: Option<String>,
    pub custom_description: Option<String>,
    pub custom_tags: Option<Vec<String>>,
}

#[derive(Debug, Serialize)]
pub struct ScrapedMetadataResponse {
    pub title: String,
    pub description: String,
    pub thumbnail_url: String,
    pub display_mode: DisplayMode,
    pub original_url: String,
    pub tags: Vec<String>,
}
